src/common/lib/util/defaults.ts

```diff
@@ -235,9 +235,14 @@ export function objectifyOptions(
   return optionsObj;
 }
 
-export function normaliseOptions(options: ClientOptions, MsgPack: MsgPack | null): NormalisedClientOptions {
+export function normaliseOptions(
+  options: ClientOptions,
+  MsgPack: MsgPack | null,
+  logger: Logger = Logger.defaultLogger, // should only be omitted by tests
```

Comment:
Yeah, I didn't do it because it would have meant updating the tests to pass a `null` value for the existing `MsgPack` argument (which they should be passing but instead are just omitting). But you're right, it's better since it means we won't forget to pass it in our TS code, so will try and change.